### ISSUE TITLE
Gene table feature [draft]

### ISSUE THREAD

**Issue by mzueva** [epam/NGB#377]
User shall be able to view the list of genes/proteins for a selected genome as a table view
![image](https://user-images.githubusercontent.com/12640002/110095931-75daff00-7dae-11eb-9bc0-adddd61a19b3.png)

#### Implementation
Add a new endpoint `POST /reference/{referenceId}/filter/gene` to `ReferenceController`
- As an input method shall accept body with the following fields (similar to `VcfFilterForm`):
  - Integer startIndex
  - Integer endIndex
  - String chromosome
  - String featureId
  - List<FeatureType> featureTypes
  - List<OrderBy> orderBy
  - Integer pageSize
  - Pointer pointer
- Method shall return `IndexSearchResult<FeatureIndexEntry>`

In short, new method shall work similar to an existing method `/reference/{referenceId}/search` and allow not only search by `featureId` but extend filtering options.

**Comment by Sergeev-Ilya** [comment]
Filter schema has some changes: I decided to use **List{String} chromosomes** instead of **String schromosomeId.**
